plugin/parser/differ/mysql/differ.go

```diff
@@ -122,22 +122,41 @@ func (*SchemaDiffer) SchemaDiff(oldStmt, newStmt string) (string, error) {
 			var alterTableInplaceDropConstraintSpecs []*ast.AlterTableSpec
 
 			oldColumnMap := buildColumnMap(oldNodes, newStmt.Table.Name)
-			for _, columnDef := range newStmt.Cols {
+			oldColumnPositionMap := buildColumnPositionMap(oldStmt)
+			for idx, columnDef := range newStmt.Cols {
 				newColumnName := columnDef.Name.Name.O
 				oldColumnDef, ok := oldColumnMap[newColumnName]
 				if !ok {
+					columnPosition := &ast.ColumnPosition{Tp: ast.ColumnPositionFirst}
+					if idx >= 1 {
```

Comment:
Add a unit test when a table has no columns for edge cases?

Reply:
If a table has no columns, it will cause error in parse stage.